File: ios/Runner/NativeButton/NativeButtonView.swift
//
//  NativeButtonView.swift
//  Runner
//
//  Created by Amit on 06/08/2025.
//

import Foundation
import UIKit
import Flutter

class NativeButtonView: NSObject, FlutterPlatformView {
    private var _view: UIView
    private let channel: FlutterMethodChannel
    
    init(
        frame: CGRect,
        viewIdentifier viewId: Int64,
        arguments args: Any?,
        binaryMessenger messenger: FlutterBinaryMessenger
    ) {
        _view = UIView(frame: frame)
        channel = FlutterMethodChannel(name: "native-button-\(viewId)", binaryMessenger: messenger)
        super.init()
        
        let button = UIButton(type: .system)
        let params = args as? [String: Any]
        let text = params?["text"] as? String ?? "Click Here"
        
        button.setTitle(text, for: .normal)
        button.frame = CGRect(x: 0, y: 0, width: 200, height: 50)
        button.addTarget(self, action: #selector(buttonPressed), for: .touchUpInside)
        
        _view.addSubview(button)
    }
    
    @objc func buttonPressed() {
        channel.invokeMethod("onButtonClick", arguments: nil)
    }
    
    func view() -> UIView {
        return _view
    }
}

File: android/app/src/main/kotlin/com/itheamc/varosa_multi_app/native_button/NativeButtonView.kt
package com.itheamc.varosa_multi_app.native_button

import android.content.Context
import android.util.Log
import android.view.View
import android.widget.Button
import io.flutter.plugin.common.BinaryMessenger
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.platform.PlatformView

class NativeButtonView(
    context: Context,
    messenger: BinaryMessenger,
    id: Int,
    creationParams: Map<*, *>?
) : PlatformView {

    private val button: Button = Button(context)
    private var _channel: MethodChannel

    init {
        val text = creationParams?.get("text") as? String ?: "Click Here"
        button.text = text
        _channel = MethodChannel(messenger, "native-button-$id")

        button.setOnClickListener {
            _channel.invokeMethod("onButtonClick", null)
        }
    }

    override fun getView(): View = button
    override fun dispose() {}
}
